### PR TITLE
implement sparsemax and sparsemax_loss

### PR DESCRIPTION
**Pull request by AndreasMadsen**
* ENH: Including a Sparsemax Later and A SparsemaxLoss class

### CODE REVIEW

File: tensorflow_addons/activations/python/sparsemax_test.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+    tf_sparsemax_op, tf_sparsemax_out = self._tf_sparsemax(z, dtype)
+
+    p0_expected = np.select([t < -1, t <= 1, t > 1], [0, (t + 1) / 2, 1])
+
+    self.assertAllCloseAccordingToType(p0_expected, tf_sparsemax_out[:, 0])
+    self.assertAllCloseAccordingToType(1 - p0_expected, tf_sparsemax_out[:, 1])
+    self.assertShapeEqual(z, tf_sparsemax_op)
+
+  def test_gradient_against_estimate(self, dtype=None):
```
Comment (seanpmorgan) [inline tensorflow_addons/activations/python/sparsemax_test.py]: I'm seeing this test fail its assertion.
```
AssertionError: 
Not equal to tolerance rtol=1e-06, atol=1e-06
Mismatched value: a is different from b. 
not close where = (array([160, 160, 160, 160, 161, 161, 161, 161, 162, 162, 162, 162, 166,
       166, 166, 166]), array([160, 161, 162, 166, 160, 161, 162, 166, 160, 161, 162, 166, 160,
       161, 162, 166]))
not close lhs = [ 0.66666667 -0.33333333 -0.33333333  0.         -0.33333333  0.66666667
 -0.33333333  0.         -0.33333333 -0.33333333  0.66666667  0.
  0.          0.          0.          0.        ]
not close rhs = [ 0.6991361  -0.3008639  -0.3008639  -0.09740829 -0.3008639   0.6991361
 -0.3008639  -0.09740829 -0.3008639  -0.3008639   0.6991361  -0.09740829
 -0.11580276 -0.11580276 -0.11580276  0.34740829]
not close dif = [0.03246943 0.03246943 0.03246943 0.09740829 0.03246943 0.03246943
 0.03246943 0.09740829 0.03246943 0.03246943 0.03246943 0.09740829
 0.11580276 0.11580276 0.11580276 0.34740829]
not close tol = [1.69913610e-06 1.30086390e-06 1.30086390e-06 1.09740829e-06
 1.30086390e-06 1.69913610e-06 1.30086390e-06 1.09740829e-06
 1.30086390e-06 1.30086390e-06 1.69913610e-06 1.09740829e-06
 1.11580276e-06 1.11580276e-06 1.11580276e-06 1.34740829e-06]
dtype = float64, shape = (170, 170)
(mismatch 0.0553633217993%)
 x: array([0., 0., 0., ..., 0., 0., 0.])
 y: array([0., 0., 0., ..., 0., 0., 0.])
```

File: tensorflow_addons/layers/__init__.py [hunk 2]
```diff
@@ -21,3 +21,4 @@
 from tensorflow_addons.layers.python.maxout import Maxout
 from tensorflow_addons.layers.python.poincare import PoincareNormalize
 from tensorflow_addons.layers.python.wrappers import WeightNormalization
+from tensorflow_addons.layers.python.sparsemax import Sparsemax
```
Comment (seanpmorgan) [inline tensorflow_addons/layers/__init__.py]: Could you please alphabetize these? We're still  working on our style guidelines

File: tensorflow_addons/losses/__init__.py [hunk 3]
```diff
@@ -19,3 +19,4 @@
 from __future__ import print_function
 
 from tensorflow_addons.losses.python.triplet import triplet_semihard_loss
+from tensorflow_addons.losses.python.sparsemax_loss import sparsemax_loss, SparsemaxLoss
```
Comment (seanpmorgan) [inline tensorflow_addons/losses/__init__.py]: Re: Alphabetize

File: tensorflow_addons/utils/python/keras_utils.py [hunk 4]
```diff
@@ -18,6 +18,7 @@
 from __future__ import print_function
 
 from tensorflow.python.keras.utils import generic_utils
+from tensorflow.python.keras.losses import LossFunctionWrapper
```
Comment (seanpmorgan) [inline tensorflow_addons/utils/python/keras_utils.py]: Re: Alphabetize

File: tensorflow_addons/utils/python/test_utils.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+# ==============================================================================
+"""Utilities for tf.test.TestCase."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import inspect
+import unittest
+
+# TODO: find public API alternative to these
```
Comment (seanpmorgan) [inline tensorflow_addons/utils/python/test_utils.py]: Good TODO. We'll be submitting a list of these API requests on TF core soon.

File: tensorflow_addons/activations/__init__.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2016 The TensorFlow Authors. All Rights Reserved.
```
Comment (seanpmorgan) [inline tensorflow_addons/activations/__init__.py]: I like breaking these out into their own subpackage. Though this directory will need a README that covers contributing guidelines (can be pretty minimal for now).

**Comment by AndreasMadsen** [comment]
Should be fixed.

File: tensorflow_addons/activations/python/sparsemax_test.py [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+
+  # calculate tau(z)
+  tau_sum = z_cumsum[np.arange(0, z.shape[0]), k_z - 1]
+  tau_z = ((tau_sum - 1) / k_z).reshape(-1, 1)
+
+  # calculate p
+  return np.maximum(0, z - tau_z)
+
+@test_utils.run_all_with_types(['float32', 'float64'])
+@test_utils.run_all_in_graph_and_eager_modes
```
Comment (facaiy) [inline tensorflow_addons/activations/python/sparsemax_test.py]: Can you use those decorator to wrap each test_* methods, rather than the class?

**Comment by facaiy** [comment]
ignore it, my mistake.

**Comment by AndreasMadsen** [comment]
Linting should be okay now.

**Pull request merged**